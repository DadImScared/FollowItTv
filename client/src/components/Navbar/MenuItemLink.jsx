
import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import MenuItem from '@material-ui/core/MenuItem';

class MenuItemLink extends Component {

  onClick = () => {
    const { handleClose, routeName } = this.props;
    handleClose(routeName);
  };

  render() {
    return (
      <MenuItem component={Link} to={this.props.routeName} onClick={this.onClick}>
        {this.props.text}
      </MenuItem>
    );
  }
}

MenuItemLink.propTypes = {
  handleClose: PropTypes.func.isRequired,
  routeName: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired
};

export default MenuItemLink;
