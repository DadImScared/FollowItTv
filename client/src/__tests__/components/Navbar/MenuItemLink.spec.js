
import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';

import MenuItem from '@material-ui/core/MenuItem';

import MenuItemLink from '../../../components/Navbar/MenuItemLink';

describe('MenuItemLink', () => {
  let props;

  beforeEach(() => {
    props = {
      handleClose: sinon.spy(),
      routeName: '/register',
      text: 'Sign up'
    };
  });

  it('should render', () => {
    const wrapper = shallow(<MenuItemLink {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleClose on click', () => {
    const wrapper = shallow(<MenuItemLink {...props} />);
    wrapper.find(MenuItem).simulate('click');
    expect(props.handleClose.calledOnce).toEqual(true);
  });
});
