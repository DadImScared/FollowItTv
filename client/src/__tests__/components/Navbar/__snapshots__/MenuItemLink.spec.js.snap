// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItemLink should render 1`] = `
<WithStyles(MenuItem)
  component={[Function]}
  onClick={[Function]}
  to="/register"
>
  Sign up
</WithStyles(MenuItem)>
`;
